tick/inference/base/learner_robust_glm.py

```diff
@@ -0,0 +1,374 @@
+# License: BSD 3 clause
+
+from warnings import warn
+import numpy as np
+from tick.base import Base
+from . import LearnerGLM
+from tick.optim.prox import ProxZero, ProxL1, ProxL2Sq, ProxElasticNet, \
+    ProxSlope, ProxMulti
+from tick.optim.solver import AGD, GD
+
+
+class LearnerRobustGLM(LearnerGLM):
+    """Learner for a Robust Generalized Linear Model (GML).
+    This is a GLM with sample intercepts (one for each sample).
+    Slope penalization is used on the individual intercepts.
+    Not intended for end-users, but for development only.
+    It should be sklearn-learn compliant.
+
+    Parameters
+    ----------
+    C_sample_intercepts : `float`
+        Level of penalization of the ProxSlope penalization used for detection
+        of outliers. Ideally, this should be equal to n_samples / noise_level,
+        where noise_level is an estimated noise level
+
+    C : `float`, default=1e3
+        Level of penalization of the model weights
+
+    fdr : `float`, default=0.05
+        Target false discovery rate for the detection of outliers, namely for
+        the detection of non-zero entries in ``sample_intercepts``
+
+    penalty : 'none', 'l1', 'l2', 'elasticnet', 'slope', default='l2'
```

Comment:
We usually present them as a set of possible values
`{‘none’, ‘l1’, ‘l2’, ‘elasticnet’, ‘slope’}, default=’l2’`

And just for me to know, why can't we use TV prox in this case?